File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/repository/StorageRepository.java
package ru.kmorozov.library.data.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import ru.kmorozov.library.data.model.book.Storage;

import java.util.List;

/**
 * Created by km on 26.12.2016.
 */
public interface StorageRepository extends MongoRepository<Storage, String> {

    Storage findByUrl(String url);

    List<Storage> findAllByParent(Storage parent);
}

File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/repository/StorageRepository.kt
package ru.kmorozov.library.data.repository

import org.springframework.data.mongodb.repository.MongoRepository
import ru.kmorozov.library.data.model.book.Storage

/**
 * Created by km on 26.12.2016.
 */
interface StorageRepository : MongoRepository<Storage, String> {

    fun findByUrl(url: String): Storage?

    fun findAllByParent(parent: Storage): MutableList<Storage>

    fun findAllByName(name: String): List<Storage>
}
